docs/docs/SM-guide/python-in-td/005-py-td-doc-strings.md

```diff
@@ -0,0 +1,200 @@
+
+
+## Examples
+
+### Function
+
+#### Practical Example
+```python
+def To_rgb_from_hex(value:str) -> tuple:
+    """Returns a color as a float tuple (rgb) converted
+    from HEX
+    
+    Args
+    ---------------
+    value (`str`)
+    > hex string to be converted into rgb 
+
+    Returns
+    ---------------
+    color (`tuple`)
+    > normalized color expressed as (r, g, b) tuple
+    """
+```
+
+#### Boiler Plate
+```python
+def Foo_bar(arg1:int) -> None:
+    """Single line description of function.
+
+    Any additional comments or operational considerations you'd like
+    to add about the function in question.
+    
+    Args
+    ---------------
+    arg1 (`int`)
+    > description of arg's role 
+
+    Returns
+    ---------------
+    None
+    """
+```
+
+#### Short Form
+The short form usage is acceptable when there is a present type hint for any arguments and returns. Provided that hints are present, the doc string can be abridged to only include a brief description of the method / function.
+
+```python
+def Foo_bar(arg1:str) -> None:
+    """One line description
+
+    Any additional comments or operational considerations you'd like
+    to add about the method in question.
+    ""
+```
+
+---
+
+### Header
+
+#### Practical Example
+```python
+"""
+SudoMagic | sudomagic.com
+Authors | Matthew Ragan, Ian Shelanskey
+Contact | contact@sudomagic.com
+"""
+
+# td python mods
+import SudoMagic
+import Lookup
+
+# pure python
+import json
+import sys
+import socket
+import logging
+```
+
+#### Boiler Plate
+```python
+"""
+SudoMagic | sudomagic.com
+Authors | Matthew Ragan, Ian Shelanskey
+Contact | contact@sudomagic.com
+"""
+
+# td python mods
+import SudoMagic
+import Lookup
+
+# pure python
+```
+
+---
+
+### Class
+
+#### Practical Example
+```python
+class Project(SudoMagic.Types.abstract_component_singleton):
+    """Project Class
+
+    The project class is responsible for the construction and distribution
+    of all extensions. _setup() constructs extensions in order, allowing
+    for a reliable and consistent start-up sequence for all python extensions
+
+    Additionally the Project class is responsible for start-up functions
+    that include loading settings from file.
+    """
+
+```
+
+#### Boiler Plate
+
+```python
+class Foo:
+    """Foo Class one line description
+
+    Any additional comments or operational considerations you'd like
+    to add about the class in question.
+    """
+```
+
+---
+
+### Method
+
+#### Practical Example
```

Comment:
The long form comment using Googles standard of "Args" and "Returns" should only be used with args and return methods that are not defined types. I find this tends to gum up the flow of the code without actaully offering that much information. 

Maybe a good way to think about these types of comments are **why** the argument exists instead of a description of it. Or at least **how** its used - and only if its not immediately obvious from the argument name.

Reply:
just consider how this looks with the vscode popup info window.